src/main/python/ttconv/scc/context.py

```diff
@@ -100,31 +103,29 @@ def get_caption_to_process(self):
 
   def has_active_caption(self) -> bool:
     """Returns whether captions are being displayed or not"""
-    return self.active_caption is not None
+    return self.active_caption.get_begin() is not None
 
   def set_buffered_caption_begin_time(self, time_code: SmpteTimeCode):
     """Initializes the current buffered caption with begin time"""
-    if self.buffered_caption is not None:
-      self.buffered_caption.set_begin(time_code)
+    self.buffered_caption.set_begin(time_code)
 
   def initialize_active_caption(self, begin_time_code: SmpteTimeCode):
     """Initializes the current active caption with id and begin time"""
-    if self.active_caption is not None:
-      if not self.active_caption.get_id():
-        self.count += 1
-        self.active_caption.set_id("caption" + str(self.count))
+    if not self.active_caption.get_id():
+      self.count += 1
```

Comment:
Shouldn't `reset_active_caption()` set the `id`?

Is this function really necessary if it only sets the `begin` time?